roles/sro_deploy_art_bundle/tasks/main.yml

```diff
@@ -0,0 +1,303 @@
+  always:
+  - name: Remove secrets from local storage
+    file:
+      path: '/tmp/{{ item }}'
+      state: absent
+    loop:
+      - authfile
+      - dockerconfigjson
+
+
+- name: Create ImageContentSourcePolicy
+  command: oc apply -f "{{ sro_imagecontentsourcepolicy_file }}"
+
+
+- name: Create namespace ci-artifacts
+  command: oc apply -f "{{ builder_namespace_file }}"
+
+
+- name: Prepare builder-dockercfg secret
+  block:
+  - name: Find the name of the builder-dockercfg secret
+    shell:
+      set -o pipefail;
+      oc get secrets -oname -n ci-artifacts
+      | cut -d/ -f2
+      | grep builder-dockercfg
+    register: builder_secret_cmd
+
+  - name: Set variable for builder_secret
+    set_fact:
+      builder_secret: "{{ builder_secret_cmd.stdout }}"
+
+  rescue:
+  - name: Failed to find the builder-dockercfg secret
+    fail: msg="No builder-dockercfg secret in the 'ci-artifacts' namespace"
+
+
+- name: Create imagestream for index and index builder
+  command: oc apply -f "{{ builder_imagestream_file }}"
+
+
+- name: Create builder script bundle-builder-helper
+  command: oc apply -f "{{ builder_script_file }}"
+
+
+- name: Create buildconfig for index builder image
+  block:
+  - name: Create build config bundle-builder-helper
+    command: oc apply -f "{{ builder_buildconfig_file }}"
+
+  - name: Wait for build config pod
+    command:
+      oc get pod/ci-artifacts-builder-1-build
+        -n ci-artifacts
+        -ojsonpath='{.status.conditions[].reason}'
+    register: buildconfig_ready
+    until: "'PodComplete' in buildconfig_ready.stdout"
+    retries: 20
+    delay: 30
+  rescue:
+  - name: "Store logs from build config"
+    shell:
+      oc logs ci-artifacts-builder-1-build
+        -n ci-artifacts
+        > "{{ artifact_extra_logs_dir }}"/failed_builder.log
+
+  - name: "Failing because of previous error"
+    fail: msg="Failing because of build config error"
+
+
+- name: Build index from latest ART bundle in a pod
+  block:
+  - name: Generate pod resource yaml
+    template:
+      src: "{{ builder_pod_file }}"
+      dest: "{{ artifact_extra_logs_dir }}/builder_pod.yml"
+      mode: 0400
+
+  - name: Create builder pod
+    command: oc apply -f "{{ artifact_extra_logs_dir }}/builder_pod.yml"
+
+  - name: Wait pod finish.
+    command:
+      oc get pod/ci-artifacts-index-builder
+        -n ci-artifacts
+        -ojsonpath='{.status.conditions[].reason}'
+    register: index_builder_ready
+    until: "'PodComplete' in index_builder_ready.stdout"
+    retries: 30
+    delay: 20
+
+  rescue:
+  - name: "Store described index builder"
+    shell:
+      oc describe pod ci-artifacts-index-builder
+        -n ci-artifacts
+        > "{{ artifact_extra_logs_dir }}"/failed_index_builder.desc
+
+  - name: "Store logs from index builder"
+    shell:
+      oc logs ci-artifacts-index-builder
+        -n ci-artifacts
+        > "{{ artifact_extra_logs_dir }}"/failed_index_builder.log
```

Comment:
can you turn this into an `always` block, so that the logs and description are always stored (mostly logs, description is less interesting when things go right)

Reply:
Done